transaction.go

```diff
@@ -458,30 +484,28 @@ func _TransactionFreezeWith(
 	client *Client,
 	body *services.TransactionBody,
 ) error {
-	if client != nil {
-		if transaction.nodeAccountIDs._IsEmpty() {
+	if transaction.nodeAccountIDs._IsEmpty() {
+		if client != nil {
 			for _, nodeAccountID := range client.network._GetNodeAccountIDsForExecute() {
 				transaction.nodeAccountIDs._Push(nodeAccountID)
 			}
-		}
 
-		if client.defaultRegenerateTransactionIDs != transaction.regenerateTransactionID {
-			transaction.regenerateTransactionID = client.defaultRegenerateTransactionIDs
+			if client.defaultRegenerateTransactionIDs != transaction.regenerateTransactionID {
+				transaction.regenerateTransactionID = client.defaultRegenerateTransactionIDs
+			}
+		} else {
+			return errNoClientOrTransactionIDOrNodeId
 		}
 	}
 
-	if transaction.nodeAccountIDs._IsEmpty() {
-		return errNoClientOrTransactionIDOrNodeId
-	}
-
-	bodyBytes, err := protobuf.Marshal(body)
-	if err != nil {
-		// This should be unreachable
-		// From the documentation this appears to only be possible if there are missing proto types
-		panic(err)
-	}
-
-	for range transaction.nodeAccountIDs.slice {
+	for _, nodeAccountID := range transaction.nodeAccountIDs.slice {
+		body.NodeAccountID = nodeAccountID.(AccountID)._ToProtobuf()
+		bodyBytes, err := protobuf.Marshal(body)
```

Comment:
is there a test case?

the license header should really go to its separate PR. it's difficult to review the changes with so much noise.

Reply:
Yes license part was my bad, it was a fast one so i thought no point of creating another pr. I added the multi_app_test and [these.](https://github.com/hashgraph/hedera-sdk-go/pull/464/commits/9cb9e978952731a9bc4bfa0c8338514a29858ce0l)